### PR TITLE
Include logging status in mavlink SYS_STATUS message

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
mavlink PR is here: https://github.com/ArduPilot/mavlink/pull/12

**Comment by peterbarker** [comment]
mavlink PR is here: https://github.com/ArduPilot/mavlink/pull/12 - CI is unlikely to work without it.

**Comment by WickedShell** [comment]
Whoooo!

**Comment by peterbarker** [comment]
On Sat, 16 Jul 2016, WickedShell wrote:

> Whoooo!

Well, it was your idea :-)

Peter

**Comment by OXINARF** [comment]
LGTM

File: libraries/DataFlash/DataFlash.cpp [hunk 1]
```diff
@@ -40,6 +40,22 @@ const struct LogStructure *DataFlash_Class::structure(uint16_t num) const
...
+    }
+    return backends[0]->logging_enabled();
+}
+bool DataFlash_Class::logging_failed() const {
+    if (_next_backend < 1) {
+        // we should not have been called!
+        return true;
+    }
+    return backends[0]->logging_failed();
+}
```
Comment (lucasdemarchi) [inline libraries/DataFlash/DataFlash.cpp]: nitpick on all these function for wrong brace placement.

Comment (peterbarker) [inline libraries/DataFlash/DataFlash.cpp]: Fixed.  Should have left them in the header :-P

**Comment by magicrub** [comment]
LGTM, @tridge please merge https://github.com/ArduPilot/mavlink/pull/12 so this can be tested on travis and merged

**Comment by WickedShell** [comment]
@tridge all CI checks are now passed (had to restart Semaphore for some reason, but thats now happy as well)

File: APMrover2/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -175,7 +184,11 @@ void Rover::send_extended_status1(mavlink_channel_t chan)
         }
     }
 
-    if (AP_Notify::flags.initialising) {
+    if (rover.DataFlash.logging_failed()) {
+        control_sensors_health &= ~MAV_SYS_STATUS_LOGGING;
+    }
+
+   if (AP_Notify::flags.initialising) {
```
Comment (lucasdemarchi) [inline APMrover2/GCS_Mavlink.cpp]: wrong indent.

File: libraries/DataFlash/DataFlash_MAVLink.h [hunk 3]
```diff
@@ -127,6 +127,10 @@ class DataFlash_MAVLink : public DataFlash_Backend
         uint8_t state_sent_max;
     } stats;
 
+    // this method is used when reporting system status over mavlink
+    bool logging_enabled() const { return true; };
```
Comment (lucasdemarchi) [inline libraries/DataFlash/DataFlash_MAVLink.h]: trailing `;`

**Comment by lucasdemarchi** [comment]
Small mistakes above. I fixed them and pushed. Thanks.

**Comment by WickedShell** [comment]
oO I admit I was looking for this to go in, but I thought the conclusion from the dev call was that some changes were going to be made for the health/enabled/present bits? I guess that can just go on top of master from now though, since what was applied is still a sane starting place.

**Pull request closed without merging**